pkg/controller/notification/notification_controller_test.go

```diff
@@ -0,0 +1,156 @@
+	apiv1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestNotificationSuccess(t *testing.T) {
+	// given
+	restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_DURATION_BEFORE_NOTIFICATION_DELETION", "10s")
+	defer restore()
+
+	t.Run("will not do anything and return requeue with shorter duration that 10s", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("jane")
+		notification := newNotification("jane", "")
+		notification.Status.Conditions = []v1alpha1.Condition{toBeDelivered()}
+		controller, request, cl := newController(t, notification, mur)
+
+		// when
+		result, err := controller.Reconcile(request)
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, result.Requeue)
+		assert.True(t, result.RequeueAfter < cast.ToDuration("10s"))
+		assert.True(t, result.RequeueAfter > cast.ToDuration("1s"))
+		murtest.AssertThatMasterUserRecord(t, "jane", cl)
+		AssertThatNotificationHasCondition(t, cl, notification.Name, toBeDelivered())
+	})
+
+	t.Run("will delete the notification as the notification duration before deletion will already pass", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("jane")
+		notification := newNotification("jane", "")
+		notification.Status.Conditions = []v1alpha1.Condition{toBeDelivered()}
+		notification.Status.Conditions[0].LastTransitionTime = v1.Time{Time: time.Now().Add(-cast.ToDuration("10s"))}
+		controller, request, cl := newController(t, notification, mur)
+
+		// when
+		result, err := controller.Reconcile(request)
+
+		// then
+		require.NoError(t, err)
+		assert.False(t, result.Requeue)
+		murtest.AssertThatMasterUserRecord(t, "jane", cl)
+		AssertThatNotificationIsDeleted(t, cl, notification.Name)
+	})
+
+}
+
+func TestNotificationDeliveredFailure(t *testing.T) {
+	restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_DURATION_BEFORE_NOTIFICATION_DELETION", "10s")
+	defer restore()
+
+	t.Run("will return an error since it cannot delete the Notification after successfully delivering", func(t *testing.T) {
+		// given
+		mur := murtest.NewMasterUserRecord("jane")
+		notification := newNotification("abc123", "")
+		notification.Status.Conditions = []v1alpha1.Condition{toBeDelivered()}
+		notification.Status.Conditions[0].LastTransitionTime = v1.Time{Time: time.Now().Add(-cast.ToDuration("10s"))}
+
+		controller, request, cl := newController(t, notification, mur)
+		cl.MockDelete = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error {
+			return fmt.Errorf("error")
+		}
+
+		// when
+		_, err := controller.Reconcile(request)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "unable to delete Notification object 'notification-name'")
```

Comment:
Check the whole error message here. Not just part of it. It's important to make sure we do include the original error (from line 84).

Also check the status. It has to be in an error state (DeletionError or whatever we call this condition).

Reply:
this is actually copy/pasted from an older pr for change tier requests... so i thought it would be fine to check part of the error. i'll make this change in both change tier requests and notifications